apps/cnquery/cmd/plugin.go

```diff
@@ -88,20 +87,6 @@ func (c *cnqueryPlugin) RunQuery(conf *run.RunQueryConfig, runtime *providers.Ru
 		return err
 	}
 
-	assetList := runtime.Provider.Connection.Inventory.Spec.Assets
-	log.Debug().Msgf("resolved %d assets", len(assetList))
-
-	assetCandidates := []*inventory.Asset{}
-	if len(assetList) > 1 && conf.PlatformId != "" {
-		filteredAsset, err := filterAssetByPlatformID(assetList, conf.PlatformId)
```

Comment:
@imilchev Afaics we need to run your new detection code before we can filter assets by platform ID, namely the part: `detectAssets connects to all assets that do not have a platform ID yet`. That's why I removed all platformID handling here and put it into `ProcessAssetCandidates` instead, so that this call results in the final assets after detection, deduplication, and platformID selection